lab/futurize.py

```diff
@@ -8,11 +8,47 @@
 import pytest
 
 
+def monkey_patch_support_for_python2():
+    from future import standard_library
+    standard_library.install_aliases()
+
+    try:
+        # Monkey patch the backported ipaddress module with
+        # alternative versions that accept string addresses, in
+        # addition to unicode addresses, in python2 code.
+        #
+        # This keep compatibility simple. Slightly complicated by the
+        # fact that some of these classes inherit from each other.
+        import ipaddress
+
+        def python2_compat(class_, bases=()):
+            def wrapper(function):
+                def __init__(self, address, *args, **kwargs):
+                    if isinstance(address, basestring):
+                        address = address.decode('utf-8')
+                    return function(self, address, *args, **kwargs)
+
+                return __init__
+
+            return type(class_.__name__, (class_,) + bases,
+                        {'__init__': wrapper(class_.__init__)})
+
+        ipaddress.IPv4Network = python2_compat(ipaddress.IPv4Network)
+        ipaddress.IPv4Address = python2_compat(ipaddress.IPv4Address)
+        ipaddress.IPv4Interface = python2_compat(ipaddress.IPv4Interface,
+                                                 bases=(ipaddress.IPv4Address,))
+
+        ipaddress.IPv6Network = python2_compat(ipaddress.IPv6Network)
+        ipaddress.IPv6Address = python2_compat(ipaddress.IPv6Address)
+        ipaddress.IPv6Interface = python2_compat(ipaddress.IPv6Interface,
+                                                 bases=(ipaddress.IPv6Address,))
+    except ImportError:
+        pass
```

Comment:
Why wouldn't it be installed?
It's a hard dep right?
Maybe a log warning instead?